weaviate/logger.py

```diff
@@ -1,5 +1,250 @@
+        print(f"DEBUG: About to log final separator: {separator}")
+        logger.debug(separator)
+        
+        print("DEBUG: Finished logging HTTP event")
+    except Exception as e:
+        print(f"DEBUG: Error in log_http_event: {str(e)}")
+        import traceback
+        print(f"DEBUG: Traceback: {traceback.format_exc()}")
+        logger.debug(f"Error logging HTTP event: {str(e)}")
+
+def log_grpc_event(method_name: str, request: Any, response: Any) -> None:
+    """Log gRPC request and response details using the default logger.
+    
+    Args:
+        method_name: The name of the gRPC method being called
+        request: The gRPC request object
+        response: The gRPC response object (may be None for request-only logging)
+    """
+    print(f"\nDEBUG: log_grpc_event called for method {method_name}")
+    print(f"DEBUG: Logger level: {logger.getEffectiveLevel()}, handlers: {logger.handlers}")
+    print(f"DEBUG: Request type: {type(request)}")
+    print(f"DEBUG: Response type: {type(response)}")
+    if logger.getEffectiveLevel() > logging.DEBUG:
+        return
+
+    try:
+        # Log request details
+        logger.debug("==================== gRPC Request ====================")
+        logger.debug(f"Method: {method_name}")
+        logger.debug("Request:")
+        try:
+            logger.debug(textwrap.indent(str(request), "  "))
+        except Exception as e:
+            logger.debug(f"  Error logging request: {e}")
+        
+        # Log response details if available
+        if response is not None:
+            logger.debug("==================== gRPC Response ===================")
+            logger.debug("Response:")
+            try:
+                logger.debug(textwrap.indent(str(response), "  "))
+            except Exception as e:
+                logger.debug(f"  Error logging response: {e}")
+            logger.debug("====================================================")
+    except Exception as e:
+        logger.debug(f"Error logging gRPC event: {str(e)}")
+
+class GrpcLoggingInterceptor(
+    grpc.aio.UnaryUnaryClientInterceptor,
+    grpc.aio.UnaryStreamClientInterceptor,
+    grpc.aio.StreamUnaryClientInterceptor,
+    grpc.aio.StreamStreamClientInterceptor
+):
+    """Interceptor that logs gRPC requests and responses."""
+
+    async def intercept_unary_unary(
+        self,
+        continuation: Callable[[grpc.aio.ClientCallDetails, Any], UnaryUnaryCall],
+        client_call_details: grpc.aio.ClientCallDetails,
+        request: Any,
+    ) -> UnaryUnaryCall:
+        """Intercept unary-unary gRPC calls."""
+        method_name = client_call_details.method
+        try:
+            # Extract just the method name from the full path for cleaner logging
+            method = method_name.split('/')[-1] if '/' in method_name else method_name
+            print(f"\nDEBUG: Intercepting gRPC call for method: {method}")
```

Comment:
This is a `print` statement and not `logger.debug` and so will show up regardless of the `WEAVIATE_LOG_LEVEL` specification making the client noisy